File: cakk-api/src/main/java/com/cakk/api/dto/param/link/ShopLinkParam.java
package com.cakk.api.dto.param.link;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import com.cakk.common.enums.LinkKind;

public record ShopLinkParam(
	@NotBlank
	LinkKind linkKind,
	@NotBlank @Size(min = 1, max = 200)
	String linkPath
) {
}

File: cakk-admin/src/main/kotlin/com/cakk/admin/dto/param/ShopLinkParam.kt
package com.cakk.admin.dto.param

import com.cakk.common.enums.LinkKind
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class ShopLinkParam(
    @field:NotBlank
    val linkKind: LinkKind,
    @field:NotBlank @field:Size(min = 1, max = 200)
    val linkPath: String
)
